### PR TITLE
i18n(es): Update of multiple pages

### PR DESCRIPTION
**Pull request by dreyfus92**
#### What kind of changes does this PR include?

- New or updated content
- Translated content

#### Description

- What does this PR change? Give us a brief description.
Updated pages based on these commits:
  - testing.mdx → 903418d7af3458dc6c7ce0dfb503ae4fd86a5921
  - typescript.mdx → b8197844a2438e2806a7302d4de0473aaccae63c
  - integrations.mdx → 633c868b17e140f193b16f2d2a8b0c554c764750
  - api-reference.mdx → d6e4cbe79b9f495a9611c29dfaa37423cd3fdfbb
  - cli-reference.mdx → ea9bc8cc89923cef54c32c38cdc11301a1cd4376
  - error-reference.mdx → efada0f8314479a4aef12184e7c2a8e683d431f9
  - integrations-reference.mdx → 6a63e6c148b86568b21cdaaab4754e04c7e96852

### CODE REVIEW

File: src/content/docs/es/guides/testing.mdx [hunk 1]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+export default defineConfig({
+  e2e: {
+    supportFile: false
+  }
+})
+```
+
+### Crea tu primer test de Cypress
+
+1. Escoge una página para probar. Este ejemplo probará la página de ejemplo `index.astro` de abajo.
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:62]: This is just a personal preference, I'm also ok with `probar`

```suggestion
1. Escoge una página para analizar. Este ejemplo analizará la página de ejemplo `index.astro` de abajo.
```

File: src/content/docs/es/guides/testing.mdx [hunk 2]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+
+    ```html title="src/pages/index.astro"
+    ---
+    ---
+    <html lang="en">
+      <head>
+        <title>¡Astro es increíble!</title>
+        <meta name="description" content="Pull content from anywhere and serve it fast with Astro's next-gen island architecture." />
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:70]: Something like this looks ok to you?

```suggestion
        <meta name="description" content="Trae contenido desde donde quieras y sírvelo rápidamente con la moderna arquitectura de islas de Astro." />
```

File: src/content/docs/es/guides/testing.mdx [hunk 3]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+      <body>
+      <h1>Hola mundo desde Astro</h1>
+      </body>
+    </html>
+    ```
+
+2. Crea un archivo `index.spec.js` en la carpeta `cypress/e2e`. Usa el siguiente test en el archivo para verificar que el título de la página y el encabezado son correctos.
+
+    ```js title="cypress/e2e/index.cy.ts"
+    it('titles are correct', () => {
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:81]:
```suggestion
    it('los titulos son correctos', () => {
```

File: src/content/docs/es/guides/testing.mdx [hunk 4]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+      const page = cy.visit('http://localhost:3000');
+      page.get('title').should('have.text', 'Astro is awesome!')
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:83]:
```suggestion
      page.get('title').should('have.text', '¡Astro es increíble!')
```

File: src/content/docs/es/guides/testing.mdx [hunk 5]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+      page.get('h1').should('have.text', 'Hello world from Astro');
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:84]:
```suggestion
      page.get('h1').should('have.text', 'Hola mundo desde Astro');
```

File: src/content/docs/es/guides/testing.mdx [hunk 6]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+    });
+    ```
+
+    :::tip[establecer un baseUrl]
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:88]:
```suggestion
    :::tip[establecer baseUrl]
```

File: src/content/docs/es/guides/testing.mdx [hunk 7]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+
+```shell
+npx cypress open
+```
+
+Una vez iniciada la App de Cypress, elige **E2E Testing** y luego selecciona el navegador que se usará para ejecutar los tests.
+
+Una vez que se haya completado la ejecución del test, deberías ver marcas de verificación verdes en la salida que confirman que tu test pasó:
+
+```shell title="Output from npx cypress run"
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:114]:
```suggestion
```shell title="Output de npx cypress run"
```

File: src/content/docs/es/guides/testing.mdx [hunk 8]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+✓ titles are correct (107ms)
+1 passing (1s)
+```
+
+:::tip[falla el test]
+Para verificar que tu test realmente funciona, puedes cambiar la siguiente línea en el archivo `index.astro`:
+
+ ```astro title="src/pages/index.astro" del={2} ins={3}
+  <body>
+    <h1>Hello world from Astro</h1>
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:125]:
```suggestion
    <h1>Hola mundo desde Astro</h1>
```

File: src/content/docs/es/guides/testing.mdx [hunk 9]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+    <h1>Hello from Astro</h1>
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:126]:
```suggestion
    <h1>Hola desde Astro</h1>
```

File: src/content/docs/es/guides/testing.mdx [hunk 10]
```diff
@@ -16,6 +16,127 @@ Es un framework de testing unitario nativo de Vite con soporte a ESM, TypeScript
...
+  </body>
+```
+
+Luego ejecuta el test de nuevo. Deberías ver una "x" roja en la salida que confirme que tu test falló.
+:::
+
+### Siguientes pasos
+
+Para más información sobre Cypress puedes encontrarla en los siguientes enlaces:
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:135]:
```suggestion
Para más información sobre Cypress puedes visitar los siguientes enlaces:
```

File: src/content/docs/es/guides/typescript.mdx [hunk 11]
```diff
@@ -84,6 +84,30 @@ import Layout from '@layouts/Layout.astro';
 }
 ```
 
+## Extending `window` and `globalThis`
```
Comment (agustinmulet) [inline src/content/docs/es/guides/typescript.mdx:87]:
```suggestion
## Extender `window` y `globalThis`
```

File: src/content/docs/es/guides/testing.mdx [hunk 12]
```diff
@@ -46,50 +167,51 @@ Alternativamente, puedes instalar Playwright dentro de tu proyecto de Astro usan
...
-  </head>
-  <body></body>
-</html>
-```
+    ```html title="src/pages/index.astro"
+    ---
+    ---
+    <html lang="en">
+      <head>
+        <title>Astro is awesome!</title>
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:177]:
```suggestion
        <title>¡Astro es increíble!</title>
```

File: src/content/docs/es/guides/testing.mdx [hunk 13]
```diff
@@ -46,50 +167,51 @@ Alternativamente, puedes instalar Playwright dentro de tu proyecto de Astro usan
...
 
-test('meta is correct', async ({ page }) => {
-  await page.goto("http://localhost:3000/");
+    test('meta is correct', async ({ page }) => {
+      await page.goto("http://localhost:3000/");
 
-  await expect(page).toHaveTitle('Astro is awesome!');
-});
-```
+      await expect(page).toHaveTitle('Astro is awesome!');
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:192]:
```suggestion
      await expect(page).toHaveTitle('¡Astro es increíble!');
```

File: src/content/docs/es/guides/testing.mdx [hunk 14]
```diff
@@ -46,50 +167,51 @@ Alternativamente, puedes instalar Playwright dentro de tu proyecto de Astro usan
...
-  <body></body>
-</html>
-```
+    ```html title="src/pages/index.astro"
+    ---
+    ---
+    <html lang="en">
+      <head>
+        <title>Astro is awesome!</title>
+        <meta name="description" content="Pull content from anywhere and serve it fast with Astro's next-gen island architecture." />
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:178]: Again, not the best translation but let me know if this is ok for you

```suggestion
        <meta name="description" content="Trae contenido desde donde quieras y sírvelo rápidamente con la moderna arquitectura de islas de Astro." />
```

File: src/content/docs/es/guides/testing.mdx [hunk 15]
```diff
@@ -46,50 +167,51 @@ Alternativamente, puedes instalar Playwright dentro de tu proyecto de Astro usan
...
 2. Crea una nueva carpeta y añade el siguiente archivo de test en `src/test`. Copia y pega el siguiente test dentro del archivo para verificar que la meta información de la página es correcta. Actualiza el valor `<title>` de la página para que coincida con la página que se está testeando.
 
-```jsx title="src/test/index.spec.ts" "Astro is awesome!"
-import { test, expect } from '@playwright/test';
+    ```jsx title="src/test/index.spec.ts" "Astro is awesome!"
+    import { test, expect } from '@playwright/test';
 
-test('meta is correct', async ({ page }) => {
-  await page.goto("http://localhost:3000/");
+    test('meta is correct', async ({ page }) => {
```
Comment (agustinmulet) [inline src/content/docs/es/guides/testing.mdx:189]:
```suggestion
    test('la metadata es corrrecta', async ({ page }) => {
```

File: src/content/docs/es/guides/typescript.mdx [hunk 16]
```diff
@@ -84,6 +84,30 @@ import Layout from '@layouts/Layout.astro';
...
+declare global {
+	var myString: string;
+	function myFunction(): boolean;
+}
+export {};
+```
+
+Esto proporcionará tipado a `globalThis.myString` y `globalThis.myFunction`, así como a `window.myString` y `window.myFunction`.
+
+Tenga en cuenta que `window` está solo disponible en el código del lado del cliente. `globalThis` está disponible tanto en el lado del servidor como en el lado del cliente, pero su valor del lado del servidor no se compartirá con el cliente.
```
Comment (agustinmulet) [inline src/content/docs/es/guides/typescript.mdx:101]:
```suggestion
Ten en cuenta que `window` sólo está disponible en el código del lado del cliente. `globalThis` está disponible tanto en el lado del servidor como en el lado del cliente, pero su valor del lado del servidor no se compartirá con el cliente.
```

File: src/content/docs/es/guides/typescript.mdx [hunk 17]
```diff
@@ -84,6 +84,30 @@ import Layout from '@layouts/Layout.astro';
...
+
+Si tu solo quieres tipar una propiedad en el objeto `window`, proporciona una interfaz `Window` en su lugar:
```
Comment (agustinmulet) [inline src/content/docs/es/guides/typescript.mdx:103]:
```suggestion
Si sólo quieres tipar una propiedad en el objeto `window`, proporciona una interfaz `Window` en su lugar:
```

**Comment by dreyfus92** [comment]
Gracias por el review @agustinmulet 🚀

**Comment by Yan-Thomas** [comment]
Thanks @dreyfus92 & @agustinmulet! We also had a few changes in the CLI page, could you please add these to the PR? https://github.com/withastro/docs/pull/2550/files#diff-3b025fbd458fbf4c168bc0d32d811f69184c6ee278c4c4bb54b81ec885df21b8

**Comment by dreyfus92** [comment]
@Yan-Thomas I've added the requested changes, thanks for the heads up. 😄

**Review by Yan-Thomas — APPROVED** [review]
Thanks @dreyfus92, @agustinmulet! Let get this merged 🚀 💜

**Pull request merged**